src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -209,5 +215,29 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler {
   uint64_t counts_[CountType::CountType_MAX] = {0};
 };
 
+class GcsResourceManagerEx : public GcsResourceManager {
```

Comment:
Can you add a description of what this class is, and why it can't just be part of `GcsResourceManager`?

Reply:
The description will be added as "GcsResourceManagerEx is responsible for updating normal task resources from node heartbeats. It is required by GCS based actor scheduling."

This class is not part of GcsResourceManager because it handles heartbeats and UpdateResourcesRequest in different ways. We could put these logic inside GcsResourceManager but that would require more if.. else.. branches. Meanwhile, we do need a single-purpose heartbeat handling function, which will be passed and called by other logic (e.g., workerLease-failure handler, which will be included in the next PR).

BTW, are you Ok with a class name ending with "Ex"?

Reply:
This logic is about updating normal task resources from node heartbeats. I've moved it to be part of GcsResourceManager.